### PR TITLE
Fixes for rust 1.67 clippy

### PR DESCRIPTION
**Pull request by seanyoung**
Signed-off-by: Sean Young <sean@mess.org>

### CODE REVIEW

**Review by xermicus — APPROVED** [review]

**Comment by LucasSte** [comment]
I haven't reviewed this PR yet, because the changes conflict with #1137 .

File: src/sema/format.rs [hunk 1]
```diff
@@ -191,7 +191,7 @@ fn parse_format_specifier(
                 Some((loc, ch)) => {
                     diagnostics.push(Diagnostic::error(
                         loc,
-                        format!("unexpected format char '{:}', expected closing '}}'", ch),
+                        format!("unexpected format char '{ch:}', expected closing '}}'"),
```
Comment (LucasSte) [inline src/sema/format.rs:194]: Is the colon still needed?

**Review by LucasSte — APPROVED** [review]
Can #1137 go in first?

**Comment by seanyoung** [comment]
> Can #1137 go in first?

Why should #1137 go in first?

**Comment by LucasSte** [comment]
> > Can #1137 go in first?
> 
> Why should #1137 go in first?

It is up to you.

**Comment by seanyoung** [comment]
> I just thought it would be difficult to review #1137, as the clippy changes won't be conspicuous in my PR.

What do you mean? They are separate changes.

**Comment by LucasSte** [comment]
> > I just thought it would be difficult to review #1137, as the clippy changes won't be conspicuous in my PR.
> 
> 
> 
> What do you mean? They are separate changes.

As I have deleted `sema/expression.rs`, your changes in the file will be lost. I'll run clippy again and fix everything. Anything I miss won't be visible, because I have moved almost all functions. I mean, Github won't show any clippy warning I missed.

**Comment by LucasSte** [comment]
> > > I just thought it would be difficult to review #1137, as the clippy changes won't be conspicuous in my PR.
> > 
> > 
> > What do you mean? They are separate changes.
> 
> As I have deleted `sema/expression.rs`, your changes in the file will be lost. I'll run clippy again and fix everything. Anything I miss won't be visible, because I have moved almost all functions. I mean, Github won't show any clippy warning I missed.

@seanyoung Feel free to merge. I don't think this is going to be a problem after all.

File: src/sema/format.rs (hunk 1 above)
Comment (seanyoung) [reply src/sema/format.rs:195]: No, that's not needed - good catch.

**Pull request merged**